OpenNefia.Content/CharaMake/CharaMakeRaceClassLayer.cs

```diff
@@ -0,0 +1,385 @@
+        [Localize] protected UiTextTopic AttributeTopic;
+        [Localize] protected UiTextTopic SkillTopic;
+        private UiVerticalContainer DetailContainer;
+        private UiVerticalContainer SkillContainer;
+        private UiVerticalContainer TrainedSkillContainer;
+        private UiWrapText DetailText;
+
+        private UiPageModel<RaceClassCell> PageModel;
+
+        private UiGridContainer AttributeContainer;
+        private UiList<RaceClass> List;
+        private RaceClassCell[] AllData;
+
+        //skills that shouldn't show up on the screen at all
+        private readonly string[] SpecialSkillIds = new[]
+        {
+            "Elona.StatLife",
+            "Elona.StatMana",
+            "Elona.StatLuck",
+            "Elona.StatSpeed"
+        };
+        
+
+        public CharaMakeRaceClassLayer()
+        {
+            Window = new UiWindow();
+            AllData = Array.Empty<RaceClassCell>();
+            PageModel = new UiPageModel<RaceClassCell>();
+
+            RaceTopic = new UiTextTopic();
+            DetailTopic = new UiTextTopic();
+            AttributeTopic = new UiTextTopic();
+            SkillTopic = new UiTextTopic();
+
+            DetailContainer = new UiVerticalContainer();
+
+            DetailText = new UiWrapText(450, UiFonts.ListTitleScreenText);
+            DetailContainer.AddElement(DetailText);
+            DetailContainer.AddLayout(LayoutType.YMin, 110);
+
+            DetailContainer.AddLayout(LayoutType.XOffset, -3);
+            DetailContainer.AddElement(AttributeTopic, LayoutType.XOffset, -7);
+
+            DetailContainer.AddLayout(LayoutType.Spacer, 18);
+
+            AttributeContainer = new UiGridContainer(GridType.Horizontal, 3, xCentered: false, xSpace: 50, ySpace: 3);
+            DetailContainer.AddElement(AttributeContainer);
+
+            DetailContainer.AddLayout(LayoutType.Spacer, 5);
+
+            DetailContainer.AddElement(SkillTopic, LayoutType.XOffset, -7);
+            SkillContainer = new UiVerticalContainer();
+            DetailContainer.AddElement(SkillContainer);
+
+            TrainedSkillContainer = new UiVerticalContainer();
+            DetailContainer.AddLayout(LayoutType.Spacer, 18);
+            DetailContainer.AddElement(TrainedSkillContainer);
+
+            OnKeyBindDown += CharaMakeRaceClassLayer_OnKeyBindDown;
+
+            List = new UiList<RaceClass>();
+            List.EventOnActivate += (_, args) =>
+            {
+                Select(args.SelectedCell.Data);
+            };
+            List.EventOnSelect += (_, args) =>
+            {
+                SelectData(args.SelectedCell.Data);
+            };
+
+            AddChild(List);
+            AddChild(Window);
+        }
+
+        public override void Initialize(CharaMakeData args)
+        {
+            base.Initialize(args);
+            AllData = GetData().Select(x => new RaceClassCell(x)).ToArray();
+
+            PageModel.OnPageChanged += Window_OnPageChanged;
```

Comment:
Nitpick: should be okay to put this callback binding in the constructor.

Reply:
done in 76f6762